### DISCUSSION TITLE
Some Draft Proposals on Omnigram (关于Omnigram的一些草案想法）

### DISCUSSION THREAD

**Discussion by exppii** [lxpio/omnigram#6] [Ideas]
## 背景

自己是一个技术控，自己有很多的资料（PDF、Epub、Word等），各种音视频，存储在NAS里面。时间一长文件和目录很混乱，自己如果要去搜索也很费劲，没法进行快速全局的搜索（类似windows 上everything）。我自己每天工作中查询的一些资料，通过浏览器浏览的内容也没有方法自动整理到本地存储和总结。阅读过很多很多书籍以后也没有在一个地方统一树立。只能随时间的流逝在脑海里也变得模糊。但是目前市面上并没有一款软件能进行统一的整理，于是我开始构建一个这样的工具，试图解决上述的问题（当然是出于自己的需求）。

## 目标和初衷：

所以Omnigram 定位是一个实现一个多终端的文档（PDF、EPUB、音视频等）管理和阅读工具（助手）。具体而言，他支持如下的一些功能（特性）：

1. 支持多终端客户端 UI（iOS、Mac、Windows、Linux、Android）；
2. 支持通过 AI 模型对文字内容进行TTS，从而实现听书功能；
3. 支持浏览器插件进行每日阅读总结并输出成文档存储为知识点（通过AI模型）；
4. 支持多种类型资料自动管理功能（NAS服务端）和全局内容搜索；
5. 对于多媒体资源如音频、视频资源提供基本的归纳和搜刮功能（类似/jellyfin视频管理功能，这里尽量是兼容和对接市面上成熟的音视频播放如infuse、kodi等软件）
6. 支持主流网盘进行加密备份功能，定期同步本地文件到如 Google Drive、One Dirve、阿里云盘等；
7. 支持基本的用户管理功能，实现家庭一些权限划分（考虑到影视分级等）；

## 目前的想法：

主要打磨阅读和听书两部分功能，后边基于此完成其他目标功能

**Comment by exppii** [comment]
## Background:

I am a tech enthusiast with a substantial collection of materials (PDFs, Epubs, Word documents, etc.) and various audio-visual content stored on a NAS. Over time, the files and directories have become disorganized, making it challenging for me to search quickly or perform a comprehensive global search (similar to 'Everything' on Windows). I struggle to organize and summarize the information I browse through in my daily work using a browser, and there's no automated method to consolidate it locally. Despite reading numerous books, I have failed to create a unified repository, and memories of the content are fading over time. As a solution to these challenges (driven by my own needs), I started building a tool. However, there is currently no software on the market that can provide unified organization. Some files are also not convenient for my children to access.

## Goals:

 Omnigram aims to be a multi-terminal document (PDF, EPUB, audio-visual, etc.) management and reading tool (assistant). Specifically, it supports the following features:

1. Multi-terminal client UI (iOS, Mac, Windows, Linux, Android).
2. Text-to-Speech (TTS) using AI models for audio content, enabling audiobook functionality.
3. Browser plugin for daily reading summaries, outputting them as documents stored as knowledge points (via AI models).
4. Automatic management of various types of materials (NAS server) and global content search.
5. Basic categorization and scraping functions for multimedia resources such as audio and video (similar to Jellyfin video management, compatible with established media players like Infuse, Kodi, etc.).
6. Encrypted backup functionality with mainstream cloud storage, periodically syncing local files to platforms like Google Drive, OneDrive, Alibaba Cloud Drive, etc.
7. User management features, allowing for certain permissions within a household (considering film and television ratings).
   
## Current:
The primary focus is on refining the reading and audiobook functionalities, with subsequent development geared towards achieving other targeted features.

**Comment by Tsuki-Zeng** [comment]
希望这个app能早日上架苹果应用商店
